src/main/java/com/marianbastiurea/lifeofbees/Apiary.java

```diff
@@ -63,24 +63,22 @@ public void splitHive(Hive hive) {
 
         List<Hive> newHives = new ArrayList<>();
 
-        if (hive.getNumberOfEggsFrame() == 6 && !hive.isItWasSplit()) {
+        if (hive.getEggsFrames().size() == 6 && !hive.isItWasSplit()) {
             System.out.println("Now old and new frames are full. Hive will be split in two hives.");
             Hive newHive = new Hive( this,this.getNumberOfHives() + 1,
-                    true, true, hive.getNumberOfHoneyFrame(), hive.getNumberOfEggsFrame(),
+                    true, true, hive.getNumberOfHoneyFrame(), hive.getEggsFrames().size(),
```

Comment:
You don't need to pass `hive.getEggsFrames().size()` anymore. Remove it.

Reply:
done